src/stores/geometryEventStore.ts

```diff
@@ -68,8 +92,25 @@ export class GeometryEventStore {
         );
     }
 
+    public undoRoutePathLink = async (event: IEvent) => {
+        const routePathLinks = event.preObject as IRoutePathLink[];
+
+        RoutePathStore!.setRoutePathLinks(routePathLinks);
+        // Need to re-initiate watcher, since we have replaced the object that was observed
+        this.observeRoutePathLinks();
+
+        if (routePathLinks.length > 0) {
+            const neighbourLinks =
+                await RoutePathLinkService.fetchAndCreateRoutePathLinksWithStartNodeId(
+                    routePathLinks[routePathLinks.length - 1].endNode.id,
+                );
+            RoutePathStore!.setNeighborRoutePathLinks(neighbourLinks);
+        } else {
+            RoutePathStore!.setNeighborRoutePathLinks([]);
+        }
+    }
```

Comment:
This method doens't seem belong to geometryEventStore. GeometryEventStore should be as generic as possible.

I know this was an old decision but I would put everything related routePaths somewhere else. What the whole logic was moved to newRoutePathLayer? I think its newRoutePath layer's responsibility to query neighborLinks when it needs

Reply:
The first implementation of this had this logic in the component, then we decided we wanted it isolated from the component. So, we refactored all the logic, and isolated it from the components.

Of course we can't have this function in this file forever. I see this as a first step, and when we get another function extracting the logic to some new layer.